coinExchange/tradeex/controllers/apiusertransmanager.py

```diff
@@ -330,7 +330,16 @@ def on_found_success_purchase_trans(api_trans):
                     notify_resp, comment)
         
         if api_trans.action == API_METHOD_PURCHASE:
+            # [Chi Question] How UserExternalWalletAddress table is used? is below logic works for separate existing tradex traffic with
+            # 3rd party api request?
+            #   My solution is to add an external_crypto_addr column for purchase order created by api user, if it's not DEFAULT_EXTERNAL_CNY_REC_ADDRESS
+            #   then it's created from 3rd party.
             external_crypto_addr = APIUserManager.get_api_user_external_crypto_addr(api_trans.api_user.user.id, 'CNY')
+
+            purchase_order = api_trans.reference_order
+            if purchase_order.external_cny_rec_address and purchase_order.external_cny_rec_address != DEFAULT_EXTERNAL_CNY_REC_ADDRESS:
```

Comment:
As such, I don't think we need this DEFAULT_EXTERNAL_CNY_REC_ADDRESS.  Either user has one record in ExternalUserWalletAddress or not.   There's no global default one.